### ISSUE TITLE
Allow disabling shellcheck checks globally within a file

### ISSUE THREAD

**Issue by austin987** [koalaman/shellcheck#777]
Disabling warnings with:
# shellcheck disable=SC1234

works fine. There are a couple checkers that I wish to ignore in my project because of a lot of false positives. In my shell test scripts, I use shellcheck -e 1234 for that, which also works great. My problem is that when using a plugin editor to run shellcheck on save, I get those warnings still. I don't want to put this in SHELLCHECK_OPTS, as I don't want to disable it globally, just in a particular file.

So, I'd to request an option to either allow
# shellcheck disable=SC1234
or
SHELLCHECK_OPTS="--exclude SC1234"

at the top of the of shell script to affect the entire file, not just the current/next line.

**Comment by koalaman** [comment]
You're in luck! In 0.4.4 and later, adding a `disable` directive at the top of the script (before any commands) works exactly as you suggest.

Please reopen if you're on 0.4.4 or later and it still doesn't do what you want.

**Comment by austin987** [comment]
@koalaman thanks for the tip. I tried with 0.4.4, and put:
#!/bin/sh

# SC2030: Modification of WINE is local (to subshell caused by (..) group).
# SC2031: WINE was modified in a subshell. That change might be lost
# shellcheck disable=SC2030,SC2031

 at the top of the file, but it's still showing errors. Should it be some other way?

Btw, could this be documented at https://github.com/koalaman/shellcheck/wiki/Ignore ? I would, but I haven't gotten it working :)

**Comment by koalaman** [comment]
There's a bug/misfeature where the directive is only recognized instead of or directly after the shebang. I filed #779 for it because that's very annoying.

In other words, this works:

```
#!/bin/sh
# shellcheck disable=SC2030,SC2031
# SC2030: Modification of WINE is local (to subshell caused by (..) group).
# SC2031: WINE was modified in a subshell. That change might be lost

true
true | WINE=42
echo $WINE
```
but this does not:


```
#!/bin/sh
# SC2030: Modification of WINE is local (to subshell caused by (..) group).
# SC2031: WINE was modified in a subshell. That change might be lost
# shellcheck disable=SC2030,SC2031

true
true | WINE=42
echo $WINE
```

Poor testing on my part. Sorry about that.

**Comment by austin987** [comment]
@koalaman, thanks, that works for me. I've subscribed to #779, looking forward to that being fixed eventually as well :).
